maintenance/util/ignore_prefix_middleware.go

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/12/17 by Charlotte Pröller
+
+package util
+
+import (
+	"log"
+	"net/http"
+	"strings"
+
+)
+
+// ConfigurableMiddleware is a wrapper for another middleware.
+// It only calls the actual middleware if none of the ignoredPrefixes is prefix of the request path
+type ConfigurableMiddleware struct {
+	ignoredPrefixes  []string
+	next             http.Handler
+	actualMiddleware http.Handler
+}
+
+// ConfigurableMiddlewareOption is ap functional Option to configure the middleware
+type ConfigurableMiddlewareOption func(*ConfigurableMiddleware) error
+
+// WithoutPrefixes allows to configure the ignoredPrefix slice
+func WithoutPrefixes(prefix ...string) ConfigurableMiddlewareOption {
+	return func(mdw *ConfigurableMiddleware) error {
+		mdw.ignoredPrefixes = append(mdw.ignoredPrefixes, prefix...)
+		return nil
+	}
+}
+
+// NewIgnorePrefixMiddleware creates a ConfigurableMiddleware
+// actualMiddleware is the actual middleware, that is called if the request is not ignored
+func NewIgnorePrefixMiddleware(next, actualMiddleware http.Handler, cfgs ...ConfigurableMiddlewareOption) *ConfigurableMiddleware {
```

Comment:
As with `ConfigurableMiddleware`: the naming should not be "middleware" as it returns a handler. As well as `actualMiddleware`.